rust/subjects/module-04.md

```diff
@@ -195,30 +207,41 @@ turn-in directory:
     ex02/
 
 files to turn in:
-    src/main.rs  Cargo.toml
+    src/lib.rs  Cargo.toml
 
 allowed symbols:
     std::fs::{metadata, Metadata, read_dir, DirEntry, ReadDir}
-    std::path::Path  std::io::{Error, Result}
-    std::env::args
+    std::path::Path
     std::{print, println, eprintln}
 ```
 
-Create a **program** that computes the total size of a directory or file. Once computed, print the size followed by a newline, and formatted like in the examples below.
+Create a **function** that computes the total size of a directory or file. Once computed, write the size followed by a newline, and formatted like in the examples below.
 
-```txt
->_ cargo run -- ~
-1.2 gigabytes
+Your function must have the following signature:
+```rust
+pub fn duh<W: std::io::Write>(writer: &mut W, basedir: &str);
+```
+
+Example Usage:
+
+```rust
+fn main() {
+    let arg: String = std::env::args().nth(1);
```

Comment:
`.nth()` returns an Option

Reply:
I might have written those mains in the markdown directly 🙈